Plugins/AWSLambdaPackager/Plugin.swift

```diff
@@ -0,0 +1,425 @@
+        }
+        return builtProducts
+    }
+
+    private func build(
+        packageIdentity: Package.ID,
+        products: [Product],
+        buildConfiguration: PackageManager.BuildConfiguration,
+        verboseLogging: Bool
+    ) throws -> [LambdaProduct: Path] {
+        print("-------------------------------------------------------------------------")
+        print("building \"\(packageIdentity)\"")
+        print("-------------------------------------------------------------------------")
+
+        var results = [LambdaProduct: Path]()
+        for product in products {
+            print("building \"\(product.name)\"")
+            var parameters = PackageManager.BuildParameters()
+            parameters.configuration = buildConfiguration
+            parameters.otherSwiftcFlags = ["-static-stdlib"]
+            parameters.logging = verboseLogging ? .verbose : .concise
+
+            let result = try packageManager.build(
+                .product(product.name),
+                parameters: parameters
+            )
+            guard let artifact = result.executableArtifact(for: product) else {
+                throw Errors.productExecutableNotFound(product.name)
+            }
+            results[.init(product)] = artifact.path
+        }
+        return results
+    }
+
+    // TODO: explore using ziplib or similar instead of shelling out
+    private func package(
+        products: [LambdaProduct: Path],
+        toolsProvider: (String) throws -> Path,
+        outputDirectory: Path,
+        verboseLogging: Bool
+    ) throws -> [LambdaProduct: Path] {
+        let zipToolPath = try toolsProvider("zip")
+
+        var archives = [LambdaProduct: Path]()
+        for (product, artifactPath) in products {
+            print("-------------------------------------------------------------------------")
+            print("archiving \"\(product.name)\"")
+            print("-------------------------------------------------------------------------")
+
+            // prep zipfile location
+            let workingDirectory = outputDirectory.appending(product.name)
+            let zipfilePath = workingDirectory.appending("\(product.name).zip")
+            if FileManager.default.fileExists(atPath: workingDirectory.string) {
+                try FileManager.default.removeItem(atPath: workingDirectory.string)
+            }
+            try FileManager.default.createDirectory(atPath: workingDirectory.string, withIntermediateDirectories: true)
+
+            // rename artifact to "bootstrap"
+            let relocatedArtifactPath = workingDirectory.appending(artifactPath.lastComponent)
+            let symbolicLinkPath = workingDirectory.appending("bootstrap")
+            try FileManager.default.copyItem(atPath: artifactPath.string, toPath: relocatedArtifactPath.string)
+            try FileManager.default.createSymbolicLink(atPath: symbolicLinkPath.string, withDestinationPath: relocatedArtifactPath.lastComponent)
```

Comment:
Code comment does not match behavior. I wonder if just renaming the executable (like the code comment says), is actually the better alternative here. What do we gain through the symlink indirection?

Reply:
I cant remember tbh, I think I tired it and it caused some issue, will need to try again. in either case its the same end result